src/Ninja/G1CP/Content/Fixes/Session/fix166_RogueWeedDialog.d

```diff
@@ -0,0 +1,94 @@
+/*
+ * #166 Ambient dialogs of rogue not available
+ */
+func int G1CP_166_RogueWeedDialog() {
+    if (!G1CP_IsFunc("Info_ORG_829_SpecialInfo_Condition", "int|none"))
+    || (!G1CP_IsIntVar("Org_829_GotJoint", 0))
+    || (!G1CP_IsInfoInst("Info_ORG_829_OfferJoint"))
+    || (!G1CP_IsItemInst("ItMiJoint_1"))
+    || (!G1CP_IsItemInst("ItMiJoint_2"))
+    || (!G1CP_IsItemInst("ItMiJoint_3")) {
+        return FALSE;
+    };
+
+    HookDaedalusFuncS("Info_ORG_829_SpecialInfo_Condition", "G1CP_166_RogueWeedDialog_Hook1");
+    HookDaedalusFuncS("Info_ORG_829_PERM_Condition", "G1CP_166_RogueWeedDialog_Hook2");
+    return TRUE;
+};
+
+/*
+ * This function intercepts the dialog condition to introduce more conditions
+ */
+func int G1CP_166_RogueWeedDialog_Hook1() {
+    G1CP_ReportFuncToSpy();
+
+    // Symbol indices (existence confirmed by function above)
+    var int joint1Id; joint1Id = MEM_GetSymbolIndex("ItMiJoint_1");
+    var int joint2Id; joint2Id = MEM_GetSymbolIndex("ItMiJoint_2");
+    var int joint3Id; joint3Id = MEM_GetSymbolIndex("ItMiJoint_3");
+    var int gotJointId; gotJointId = MEM_GetSymbolIndex("Org_829_GotJoint");
+
+    // Add the new conditions (other conditions remain untouched)
+    var int cond1;
+    var int cond2;
+    var int cond3;
+    var int cond4;
+
+    // Check if dialog was told (check if symbol exists first!)
+    cond1 = !Npc_KnowsInfo(hero, MEM_GetSymbolIndex("Info_ORG_829_OfferJoint"));
+
+    // Check if NPC has a joint
+    cond2 = !Npc_HasItems(self, joint1Id);
+    cond3 = !Npc_HasItems(self, joint2Id);
+    cond4 = !Npc_HasItems(self, joint3Id);
+
+    // Return false if either of the conditions is true
+    if (cond1) || (cond2 && cond3 && cond4) {
```

Comment:
This if-condition might cause issues. It will prevent the dialog from appearing if the variable is `Org_829_GotJoint` is true but the NPC no longer has the items. We will have to revise this if-condition. We can brain storm some ideas here.

Since we are adding an or-condition, it is dangerous to prematurely return false. Likewise, it is dangerous to return true, because a mod could have added new conditions in the dialog condition function.

I wonder if it will be enough to just change the order: Run the original function first (use `ContinueCall()` in the beginning and collect its return value). If the return value is true, return true, if it is false, add our new conditions and return true/false. If I am not mistaken we might have done this in a few fixes. Maybe we can find those dialog fixes.

Reply:
Actually this is how we handles these types of issues in the past (see #16 for reference). We should revisit these fixes as well then.

Reply:
I agree with the comment about #66. That may be worth worrying about, if the dialog was an important one and is likely to be changed in a mod. Here - you are right - there is no need to overthink it. 

Nevertheless, the issue I mentioned in the first paragraph in the comment above should be addressed. And I think your suggestion with #96 sounds great. 

What we should do is check if the variable that is used to check whether the joint has been given is false. If so, check if the dialog was told and the NPC has either of the items and then set the variable to true. After that (irrespective of the check of the variable) the original function should be triggered.